typegate/tests/prisma/self_relationship_test.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright Metatype OÜ under the Elastic License 2.0 (ELv2). See LICENSE.md for usage.
+
+import { gql, recreateMigrations, test } from "../utils.ts";
+
+test("1:n relationships", async (t) => {
+  const tgPath = "prisma/self_relationship.py";
+  const e = await t.pythonFile(tgPath);
+
+  await t.should("drop schema and recreate", async () => {
+    await gql`
+      mutation a {
+        dropSchema
+      }
+    `
+      .expectData({
+        dropSchema: 0,
+      })
+      .on(e);
+    await recreateMigrations(e);
+  });
+
+  await t.should("insert an isolated node", async () => {
+    await gql`
+      mutation q {
+        createNode(data: { name: "One" }) {
+          name
+        }
+      }
+    `
+      .expectData({
+        createNode: { name: "One" },
+      })
+      .on(e);
+  });
+
+  await t.should("append a node", async () => {
+    await gql`
+      mutation q {
+        createNode(data: {
+          name: "Two",
+          prev: {
+            connect: {
+              name: "One"
+            }
+          }
+        }) {
+          name
+        }
+      }
+    `
+      .expectData({
+        createNode: {
+          name: "Two",
+        },
+      })
+      .on(e);
+  });
+
+  await t.should("insert a list", async () => {
+    // This does not work!
+    // Should this work??
```

Comment:
I would say yes!

Reply:
I will see after we merge again the full prisma mapping...